src/win/util.c

```diff
@@ -1663,6 +1673,127 @@ int uv_os_unsetenv(const char* name) {
 }
 
 
+static int uv__gethostnamew_nt60(WCHAR* name, int name_len) {
+  int result_len;
+  int error_code = NO_ERROR;
+
+  /* WSALookupService stuff
+   * Avoid dynamic memory allocation if possible */
+  CHAR local_buf[WSAQ_LOCAL_BUF_LEN];
+  DWORD dwlen = (DWORD)WSAQ_LOCAL_BUF_LEN;
+  WSAQUERYSETW* pwsaq;
+  /* hostname returned from WSALookupService stage */
+  WCHAR* result_name = NULL;
+  /* WSALookupService handle */
+  HANDLE hlookup;
+  /* Fallback to heap allocation if stack buffer is too small */
+  WSAQUERYSETW* heap_data = NULL;
+
+  /* check input */
+  if (name == NULL) {
+    error_code = WSAEFAULT;
+    goto cleanup;
+  }
+
+  /* 
+   * Stage 1: Check environment variable
+   * _CLUSTER_NETWORK_NAME_ len == ComputeName(NETBIOS) len.
+   * i.e 15 characters + null.
+   * It overrides the actual hostname, so application can
+   * work when network name and computer name are different 
+   */ 
+  result_len = GetEnvironmentVariableW(L"_CLUSTER_NETWORK_NAME_",
+                                       name,
+                                       name_len);
+  if (result_len != 0) {
+    if (result_len > name_len) {
+      error_code = WSAEFAULT;
+    }
+    goto cleanup;
+  }
+
+  /* Stage 2: Do normal lookup through WSALookupServiceLookup */
+  pwsaq = (WSAQUERYSETW*)local_buf;
```

Comment:
style nit: space before cast
```suggestion
  pwsaq = (WSAQUERYSETW*) local_buf;
```